src/components/Settings.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useState } from 'react';
+import {
+  Accordion,
+  AccordionSummary,
+  Typography,
+  AccordionDetails,
+  Box,
+  Snackbar,
+  Button,
+  TextField,
+  makeStyles,
+  FormControl,
+  FormLabel,
+  RadioGroup,
+  FormControlLabel,
+  Radio,
+  Collapse,
+} from '@material-ui/core';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import { Alert } from '@material-ui/lab';
+import { useSettings, AuthMethod } from '../helpers/ContextProvider';
+
+const useStyles = makeStyles(theme => ({
+  tabPanelRoot: {
+    '& > * + *': {
+      marginTop: theme.spacing(2),
+    },
+  },
+}));
+
+const Settings: React.FC = () => {
+  const classes = useStyles();
+  const [settings, saveSettings] = useSettings();
+
+  const [apiKey, setApiKey] = useState(settings.apiKey);
+  const [authMethod, setAuthMethod] = useState(settings.authMethod);
+  const [project, setProject] = useState(settings.project);
+
+  const [saved, setSaved] = useState(false);
+
+  return (
+    <>
+      <Snackbar
+        autoHideDuration={1000}
+        open={saved}
+        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
+        onClose={() => setSaved(false)}
+      >
+        <Alert severity="success">Settings saved in local storage.</Alert>
+      </Snackbar>
+      <Accordion style={{ maxWidth: '400px' }}>
```

Comment:
I'm not too keen on the accordion if I'm honest, though I take your point about using Modals. I wonder if it'd make sense to have a settings "page" - perhaps a route under our plugin root. I'd like to have an approach that all our plugins could follow which would create consistency for us and for the user though having said that ideally that would apply for all backstage plugins. 
Anyway, this is fine for now, don't worry, I'm just thinking out loud.

Reply:
Accordion wins over modal in the fact that it doesn't force context on user - he can open the settings, look back at the data table, save the settings, look back another time to check if the data fetched correctly and then hide the settings. I know that Accordion is meant rather for text and it looks strange to not be full-width (but for me both narrow and centered and full-width looks worse)
Maybe there is some better component to show and hide settings panel?